lib/sycamore/sycamore/transforms/extract_document_structure.py

```diff
@@ -16,6 +18,50 @@ def extract(document: Document) -> Union[Document, HierarchicalDocument]:
         pass
 
 
+class StructureByImages(DocumentStructure):
+    @staticmethod
+    def extract(doc: Document) -> HierarchicalDocument:
+        doc = HierarchicalDocument(doc.data)
+        doc.data["relationships"] = doc.get("relationships", {})
+        doc.data["label"] = doc.get("label", "DOCUMENT")
+
+        images = [child for child in doc.children if child.type == "Image"]
+        doc.children = []
+        for image in images:
+            initial_page = HierarchicalDocument(
```

Comment:
You're gonna create a 'front page' doc for every image in a document? That seems not right.

Reply:
yeah this was because of how we organize a graph pipeline. We iterate over the children of each document. each child has a summary. So we attach images to each section(or front page doc), summarize each section(base64 encode the images), and then once its passed to openai that "summary" is passed in. I know it's weird but for this im not sure what a better abstraction would be that would support all types of summaries naturally.